File: Packages/Flip54Core/Sources/Flip54Core/Equipment.swift
public struct Equipment: Codable, Hashable, Sendable {
    public var hasWeights: Bool
    public var hasPullUpBar: Bool
    public var hasYogaMat: Bool

    public init(hasWeights: Bool, hasPullUpBar: Bool, hasYogaMat: Bool) {
        self.hasWeights = hasWeights
        self.hasPullUpBar = hasPullUpBar
        self.hasYogaMat = hasYogaMat
    }

    public static let bodyWeightOnly = Equipment(hasWeights: false, hasPullUpBar: false, hasYogaMat: false)
    public static let fullKit        = Equipment(hasWeights: true,  hasPullUpBar: true,  hasYogaMat: true)
    public static let barOnly        = Equipment(hasWeights: false, hasPullUpBar: true,  hasYogaMat: false)
    public static let weightsOnly    = Equipment(hasWeights: true,  hasPullUpBar: false, hasYogaMat: false)
}

File: android/core/src/main/kotlin/com/flip54/core/Equipment.kt
package com.flip54.core

import kotlinx.serialization.Serializable

@Serializable
data class Equipment(
    val hasWeights: Boolean,
    val hasPullUpBar: Boolean,
    val hasYogaMat: Boolean
) {
    companion object {
        val bodyWeightOnly = Equipment(hasWeights = false, hasPullUpBar = false, hasYogaMat = false)
        val fullKit        = Equipment(hasWeights = true,  hasPullUpBar = true,  hasYogaMat = true)
        val barOnly        = Equipment(hasWeights = false, hasPullUpBar = true,  hasYogaMat = false)
        val weightsOnly    = Equipment(hasWeights = true,  hasPullUpBar = false, hasYogaMat = false)
    }
}
